deepchem/models/torch_models/infograph.py

```diff
@@ -101,93 +293,169 @@ class InfoGraphModel(ModularTorchModel):
-        self.average_loss = average_loss
+        self.task = task
+        if self.task == 'supervised':
+            self.embedding_dim = embedding_dim
+        elif self.task == 'semisupervised':
+            self.embedding_dim = embedding_dim * num_gc_layers
+            self.num_gc_layers = num_gc_layers
 
         self.components = self.build_components()
         self.model = self.build_model()
         super().__init__(self.model, self.components, **kwargs)
+        self.MIloss = MolGraphConvMILoss(self.device, measure, average_loss)
 
     def build_components(self):
-        return {
-            'encoder':
-                InfoGraphEncoder(self.num_features, self.edge_features,
-                                 self.embedding_dim),
-            'unsup_encoder':
-                InfoGraphEncoder(self.num_features, self.edge_features,
-                                 self.embedding_dim),
-            'ff1':
-                MultilayerPerceptron(2 * self.embedding_dim, self.embedding_dim,
-                                     (self.embedding_dim,)),
-            'ff2':
-                MultilayerPerceptron(2 * self.embedding_dim, self.embedding_dim,
-                                     (self.embedding_dim,)),
-            'fc1':
-                torch.nn.Linear(2 * self.embedding_dim, self.embedding_dim),
-            'fc2':
-                torch.nn.Linear(self.embedding_dim, 1),
-            'local_d':
-                MultilayerPerceptron(self.embedding_dim,
-                                     self.embedding_dim, (self.embedding_dim,),
-                                     skip_connection=True),
-            'global_d':
-                MultilayerPerceptron(2 * self.embedding_dim,
-                                     self.embedding_dim, (self.embedding_dim,),
-                                     skip_connection=True),
-        }
+        """
+        Builds the components of the InfoGraphStar model. InfoGraphStar works by maximizing the mutual information between the graph-level representation and the representations of substructures of different scales.
+
+        It does this by producing graph-level encodings and substructure encodings, and then using a discriminator to classify if they are from the same molecule or not.
+
+        The encoder is a graph convolutional network that produces the graph-level encodings and substructure encodings.
+
+        In a supervised training mode, only 1 encoder is used and the encodings are not compared, while in a semi-supvervised training mode they are different in order to prevent negative transfer from the pretraining stage.
+
+        The local discriminator is a multilayer perceptron that classifies if the substructure encodings are from the same molecule or not while the global discriminator classifies if the graph-level encodings are from the same molecule or not.
+
+        Components list and type:
+        ------------------------
+        Encoder: InfoGraphEncoder
```

Comment:
These names should match the exact strings used internally. I think we will later want to expand the modular torch model API to allow fetching the outputs of particular components